mod_app/models/film_model.py

```diff
@@ -87,10 +85,17 @@ def __str__(self):
         ("other", "Other"),
     }
     format_type = models.CharField(
-        max_length=5, default="other", verbose_name="format", choices=FORMAT_CHOICES
+        max_length=5,
+        default="other",
+        verbose_name="format",
+        choices=FORMAT_CHOICES,
     )
     format_other = models.CharField(
-        max_length=255, blank=True, null=True, help_text="Use this if you chose 'other'"
+        max_length=255,
+        blank=True,
+        null=True,
+        help_text="Use this if you chose 'other'",
+        validators=[validate_format_other],
```

Comment:
I think this validator needs to be attached to format_type. Not sure if it'll be possible though this way. Maybe you need to override the clean method on the form?

Reply:
I think I fixed this in a later PR - I attached it to the save method

Reply:
I think a better option might be to attach a clean() method to the model:
https://docs.djangoproject.com/en/4.2/ref/models/instances/#validating-objects

That way the admin form should use it to validate the data and display a helpful error message.

Reply:
have done on the bib branch 👍